src/main/java/ml/compute/InputData.java

```diff
@@ -0,0 +1,28 @@
+package ml.compute;
+
+import java.util.Set;
+
+import ml.preprocessor.types.PreprocessedData;
+import ml.preprocessor.types.RawData;
+
+/**
+ * Created by mikeecb on 17/10/2015.
+ */
+public class InputData {
```

Comment:
I was thinking this class to be more abstract than this. Kind of like either type. I don't think this implementation is a good idea, because usually you want just the raw data and do not care about preprocessed one and vice versa.